### PR TITLE
Add modules support for azure

### PR DESCRIPTION
**Pull request by clebs**
**Description**

Changes proposed in this pull request:

- Modules support for azure
- Providers without modules support switch to using old templates.

**Related issue(s)**
#28

### CODE REVIEW

File: provision/examples/azure/README.md [hunk 1]
```diff
@@ -10,9 +10,11 @@ This example shows you how to use Hydroform to provision an Azure cluster.
 
 To provision an Azure cluster you need:
 
-1. The **Client ID** and **Client Secret** of your Azure subscription stored in a TOML file:
+1. The **Subscription ID**, **Tenant ID**, **Client ID** and **Client Secret** of your Azure subscription stored in a TOML file:
```
Comment (bszwarc) [inline provision/examples/azure/README.md]:
```suggestion
1. The **Subscription ID**, **Tenant ID**, **Client ID**, and **Client Secret** of your Azure subscription stored in a TOML file:
```

**Comment by clebs** [comment]
Thanks for pointing the issue out @suleymanakbas91! Should be fixed now.

File: provision/examples/azure/main.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func main() {
 	log.SetOutput(ioutil.Discard)
 
 	cluster := &types.Cluster{
-		KubernetesVersion: "1.15",
+		KubernetesVersion: "1.15.7",
```
Comment (rakesh-garimella) [inline provision/examples/azure/main.go]: I would like 1.16 here since we kyma supports 1.16 already

Comment (clebs) [reply provision/examples/azure/main.go]: Done thanks!

**Comment by a-thaler** [comment]
As @suleymanakbas91 is on vacation I will force-merge as all feedback is addressed.

**Pull request merged**